slither/solc_parsing/declarations/function.py

```diff
@@ -770,6 +770,7 @@ def _parse_variable_definition(self, statement: Dict, node: NodeSolc) -> NodeSol
                     for v in variables:
                         identifier = {
                             "nodeType": "Identifier",
+                            "id": v["id"],
```

Comment:
Declaration ASTs have the "id" attribute, which Identifier ASTs have the "referencedDeclaration".
Here, slither is creating synthetic Identifier AST nodes for the generated initializer statement. Thus, we should use "referencedDeclaration" attribute here.

Reply:
BTW, this is a bug in slither. I believe this is a correct fix. Let's submit a PR to them.

Reply:
I suspect that the declaration may not have "id". In that case, we shouldn't add "referencedDeclaration". But, this is just a speculation. Can you confirm tuple variable decls always have "id"?

From variable_declaration.py,
```
        # Only for comapct ast format
        # the id can be used later if referencedDeclaration
        # is provided
        if "id" in var:
            self._reference_id = var["id"]
```

Reply:
In the current version of solc, all variable declarations have `id`. They are written to json using the `ASTJsonExporter::visit(VariableDeclaration const& node)` function, which writes `id` by calling `setJsonNode`. It's possible that past versions didn't set `id`, but I tested the example out using 0.5.13 and slither didn't crash.

Reply:
Crytic wants us to sign a contributor license agreement before we submit any PRs to them:
```
Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/crytic/slither?pullRequest=1507) before we can accept your contribution.
0 out of 3 committers have signed the CLA.

❌ Runming Li
❌ duckki
❌ kevinclancy
```